### ISSUE TITLE
Not Receiving Rapid Messages

### ISSUE THREAD

**Issue by slalomchip** [miguelgrinberg/python-socketio#662]
I’m having an issue receiving multiple messages that are rapidly sent.  

Multiple clients register themselves with a server for a particular event. Clients emit messages to the server and the server broadcasts the message to all other clients.  The client registers itself with the server by sending the `manual_connection_parameter` message.  Once the `connect_confirm` message is received, registration with the server is complete.

One client sends up to about 20 `start_list` messages in rapid succession (all within one second). The longest length of a `start_list` message in the test data is 854.  The code below receives the first message and then hangs. I am aware of other clients (not Python) that receive all the messages without any issue.

My script runs on a Windows 10 machine.  The server is running Socket.IO protocol 3.x so I am using python-socketio v 4.6.1 python-engineio v3.14.2.  

Should python-socketio be able to receive 20-30 messages within one second?  Or is there a problem with the code below?

```
import socketio, json

sio = socketio.Client()  
url = 'http://MyServerCOnnection.com:5000/'
isConnected = True

# Handshake
try:
    sio.connect(url)
except:
    isConnected = False
if isConnected == True:  
    data = json.dumps('eventid': '12345',
            'application': 'MyApp',
            'version': '1.0.0')         
    sio.emit('manual_connection_parameter', data)

@sio.event
def connect_confirm(data):
    data = json.loads(data)
    print('\nConnect Confirm = ', data)
    
@sio.event
def start_list(data):
    data = json.loads(data)
    print('\nStart List = ', data)

sio.wait()
```

**Comment by miguelgrinberg** [comment]
Looks in the troubleshooting section of the docs for instructions on how to enable logging. Then please show me a logging session.

**Comment by slalomchip** [comment]
Here is the logged data.  I removed the print statements from my script.  I also changed the URL in the log because it isn't my server.

Attempting polling connection to http://ewscdata.com:40000/socket.io/?transport=polling&EIO=3
Polling connection accepted with {'sid': '6ndgVk4QqHGGmp2GAAQM', 'upgrades': ['websocket'], 'pingInterval': 25000, 'pingTimeout': 5000}
Engine.IO connection established
Received packet MESSAGE data 0
Namespace / is connected
websocket-client package not installed, only polling transport is available
Sending packet PING data None
Sending polling GET request to http://ewscdata.com:40000/socket.io/?transport=polling&EIO=3&sid=6ndgVk4QqHGGmp2GAAQM
Emitting event "manual_connection_parameter" [/]
Sending packet MESSAGE data 2["manual_connection_parameter","{\"loggingdetail\": \"no\", \"mode\": \"Tournament\", \"eventid\": \"20C098\", \"eventsubid\": \"\", \"provider\": \"Johnson\", \"application\": \"TTM\", \"version\": \"1.2.0\", \"username\": \"myemail@cox.net\"}"]
Received packet PONG data None
Received packet MESSAGE data 2["connect_confirm","{\"eventId\":\"20C098\",\"eventSubId\":\"\",\"pin\":\"8516\"}"]
Received event "connect_confirm" [/]
Sending polling GET request to http://ewscdata.com:40000/socket.io/?transport=polling&EIO=3&sid=6ndgVk4QqHGGmp2GAAQM
Received packet MESSAGE data 2["start_list","{\"startlistName\":\"EventGroup B5\",\"round\":0,\"eventName\":\"Jump\",\"division\":\"B5\",\"group\":\"B5\",\"lake\":\"\",\"current_athlete_index\":0,\"current_athlete_name\":\"\",\"startlist_athletes\":[{\"athleteId\":\"400163265\",\"athleteName\":\"Smith, Levi\",\"athleteCountry\":\"USA\",\"athleteRegion\":\"IN\",\"athleteDivision\":\"B5\",\"athleteGroup\":\"B5\",\"position_starting\":1,\"position_seed\":26,\"position_current\":1,\"score_seed\":0.000,\"score_current\":0.0}]}"]
Received event "start_list" [/]
Sending polling GET request to http://MyConnectionURL.com:5000/socket.io/?transport=polling&EIO=3&sid=6ndgVk4QqHGGmp2GAAQM
Unexpected packet from server, aborting
Waiting for write loop task to end
Exiting write loop task
Waiting for ping loop task to end
Sending packet PING data None
PONG response has not been received, aborting
Exiting ping task
Engine.IO connection dropped
Connection failed, new attempt in 1.42 seconds
Exiting read loop task
Attempting polling connection to http://ewscdata.com:40000/socket.io/?transport=polling&EIO=3
Polling connection accepted with {'sid': '_JzXegNiGVSZ1pQuAAQN', 'upgrades': ['websocket'], 'pingInterval': 25000, 'pingTimeout': 5000}
Engine.IO connection established
Received packet MESSAGE data 0
Namespace / is connected
websocket-client package not installed, only polling transport is available
Sending packet PING data None
Sending polling GET request to http://MyConnectionURL.com:5000/socket.io/?transport=polling&EIO=3&sid=_JzXegNiGVSZ1pQuAAQN
Reconnection successful
Received packet PONG data None
Sending polling GET request to http://MyConnectionURL.com:5000/socket.io/?transport=polling&EIO=3&sid=_JzXegNiGVSZ1pQuAAQN

**Comment by miguelgrinberg** [comment]
> websocket-client package not installed, only polling transport is available

This is going to severely impact the performance of your client. Try adding the WebSocket client.

**Comment by slalomchip** [comment]
That appears to have fixed it!  The owner of the server told me that the older protocols were being used, but I just now upgraded to the newest python-socketio and python-engineio packages and I also installed websocket-client.  I successfully received all the messages.  I'll check out all the other messages thoroughly - hopefully I can stay on the latest version of python-socketio and python-engineio.  Thanks!
